addons/GearSwap/libs/closetCleaner.lua

```diff
@@ -0,0 +1,367 @@
+--Copyright (c) 2016-2017, Brimstone
+--All rights reserved.
+
+--Redistribution and use in source and binary forms, with or without
+--modification, are permitted provided that the following conditions are met:
+
+--    * Redistributions of source code must retain the above copyright
+--      notice, this list of conditions and the following disclaimer.
+--    * Redistributions in binary form must reproduce the above copyright
+--      notice, this list of conditions and the following disclaimer in the
+--      documentation and/or other materials provided with the distribution.
+--    * Neither the name of closetCleaner nor the
+--      names of its contributors may be used to endorse or promote products
+--      derived from this software without specific prior written permission.
+
+--THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+--ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+--WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+--DISCLAIMED. IN NO EVENT SHALL Brimstone BE LIABLE FOR ANY
+--DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+--(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+--LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+--ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+--(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+--SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+-- _addon.name = 'closetCleaner'
+-- _addon.version = '0.9'
+-- _addon.author = 'Brimstone'
+-- _addon.commands = {'cc','closetCleaner'}
+
+local cc = {}
+cc.sandbox = {}
+cc.sandbox.windower = setmetatable({}, {__index = windower})
+cc.sandbox.windower.register_event = functions.empty()
```

Comment:
Are you trying to disable `register_event`? If so, you'd want `functions.empty`, not `functions.empty()`. The latter is the same as using `nil`.

Any particular reason to do that? I would probably throw an error instead, so people actually know not to use it, instead of nothing happening:
```lua
cc.sandbox.windower.register_event = error:apply('Registering events disabled for CC')
```

Reply:
The error might lead to confusion when people run the script for the first time. 

The library only loads the user files to get to the gear sets, but `register_event` (and a few other functions, I think?) would make persistent changes within the environment of the user's current gearswap file. If they weren't blocked, you'd have to reload gearswap after running the script, but there's no real need to know that the library is disabling them within its own sandbox.